packages/canary-web-components/src/components/ic-data-table-title-bar/ic-data-table-title-bar.tsx

```diff
@@ -0,0 +1,149 @@
+  IcDensityUpdateEventDetail,
+} from "../ic-data-table/ic-data-table.types";
+import { isSlotUsed } from "../../utils/helpers";
+
+const DEFAULT_TITLE_BAR_HEADING = "Title Bar";
+
+/**
+ * @slot primary-action - Render an interactive element that will perform a primary action. Renders to the right of the density select.
+ * @slot custom-actions - Render additional custom actions to the left of the density select.
+ * @slot heading - The heading to render on the data table title bar.
+ * @slot description - Render an alternative description in the description section.
+ */
+@Component({
+  tag: "ic-data-table-title-bar",
+  styleUrl: "ic-data-table-title-bar.css",
+  shadow: true,
+})
+export class DataTableTitleBar {
+  private initialDensitySelectOption: IcDataTableDensityOptions = "default";
+  private isEmbedded: boolean = false;
+
+  @Element() el: HTMLIcDataTableTitleBarElement;
+
+  /**
+   * The description that is displayed below the `heading` and `metadata`. Can be overridden with the `description` slot.
+   */
+  @Prop() description?: string;
+
+  /**
+   * The heading of the title bar. Can be overridden with the `heading` slot. If used with an ic-data-table it will default to the table's `caption` unless overridden.
+   */
+  @Prop() heading?: string = DEFAULT_TITLE_BAR_HEADING;
+
+  /**
+   * When `true`, the density select will not be rendered.
+   */
+  @Prop() hideDensitySelect?: boolean = false;
+
+  /**
+   * The metadata displayed next to the `heading`.
+   */
+  @Prop() metadata?: string;
+
+  /**
+   * Emitted when the table density select value is changed.
+   */
+  @Event() icTableDensityUpdate: EventEmitter<IcDensityUpdateEventDetail>;
+
+  componentWillLoad(): void {
+    const parentEl = this.el.parentElement;
+    if (parentEl.tagName === "IC-DATA-TABLE") {
+      const dataTable = parentEl as HTMLIcDataTableElement;
+      this.initialDensitySelectOption = dataTable.density;
+      this.isEmbedded = dataTable.embedded;
+      if (this.heading === DEFAULT_TITLE_BAR_HEADING) {
+        this.heading = dataTable.caption;
+      }
+    }
+  }
+
+  private changeDensity = (
+    ev: IcSelectCustomEvent<IcDensityUpdateEventDetail>
+  ) => {
+    this.icTableDensityUpdate.emit({ value: ev.detail.value });
+  };
+
+  render() {
+    const {
+      changeDensity,
+      description,
+      el,
+      heading,
+      hideDensitySelect,
+      initialDensitySelectOption,
+      isEmbedded,
+      metadata,
+    } = this;
+    const showActionArea =
+      isSlotUsed(el, "primary-action") ||
+      isSlotUsed(el, "custom-actions") ||
+      !hideDensitySelect;
+
+    const hasDescription = isSlotUsed(el, "description") || description;
+    return (
+      <div
+        class={{
+          "title-bar-wrapper": true,
+          "full-width": isEmbedded,
+        }}
+      >
+        <div class="header-container">
+          <slot name="heading">
+            <ic-typography variant="h3">{heading}</ic-typography>
```

Comment:
Currently the default heading isn't actually an `<h3>` (i checked out the [Storybook](https://mi6.github.io/ic-ui-kit/branches/canary/data-tables-v2-new/canary-web-components/?path=/story/web-components-data-table-title-bar--with-metadata-and-description)) 

You'll need to set it as an h3 like so:
`<ic-typography variant="h3"><h3>{heading}</h3></ic-typography>`

Reply:
i'm gonna raise this as a bug against typography code examples to make it clearer that this needs to be done